roachtest: sqlsmith/setup=rand-tables/setting=no-mutations failed

roachtest.sqlsmith/setup=rand-tables/setting=no-mutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5419500&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5419500&tab=artifacts#/sqlsmith/setup=rand-tables/setting=no-mutations) on release-21.2 @ [6fb1297da998e4b3e231996074ddc27a2cbb7a6a](https://github.com/cockroachdb/cockroach/commits/6fb1297da998e4b3e231996074ddc27a2cbb7a6a):


```
The test failed on branch=release-21.2, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=rand-tables/setting=no-mutations/run_1
	sqlsmith.go:223,sqlsmith.go:256,test_runner.go:777: error: pq: internal error: crdb_internal.revalidate_unique_constraints_in_all_tables(): concurrent txn use detected. ba: Scan [/Table/3/1,/Table/3/2), [txn: ea4877d6], [can-forward-ts]
		stmt:
		WITH
			with_25296 (col_145659, col_145660, col_145661)
				AS (
					SELECT
						*
					FROM
						(VALUES (4072727462:::OID, NULL, NULL), (3548358701:::OID, '-Inf':::FLOAT8, 3098850418:::OID))
							AS tab_63732 (col_145659, col_145660, col_145661)
				),
			with_25297 (col_145662)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								(NULL),
								((-1184669663):::INT8),
								(CASE WHEN true THEN 2147483647:::INT8 ELSE 2020693423:::INT8 END)
						)
							AS tab_63733 (col_145662)
				)
		SELECT
			(-18401):::INT8 AS col_145663,
			tab_63736.col2_3 AS col_145664,
			'2021-06-06 02:42:11.000574':::TIMESTAMP AS col_145665
		FROM
			with_25297 AS cte_ref_7219,
			defaultdb.public.table2@table2_col2_7_col2_9_col2_8_col2_1_col2_6_col2_4_col2_2_col2_5_col2_3_col2_0_col2_10_key
				AS tab_63736
			JOIN defaultdb.public.table2@[0] AS tab_63737 ON (tab_63736.col2_7) = (tab_63737.col2_7),
			with_25296 AS cte_ref_7220
		WHERE
			crdb_internal.revalidate_unique_constraints_in_all_tables()::BOOL
		LIMIT
			21:::INT8;
```
<details><summary>Reproduce</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=rand-tables/setting=no-mutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/internal/sqlsmith/schema.go
+++ b/pkg/internal/sqlsmith/schema.go
@@ -504,6 +504,7 @@ var functions = func() map[tree.FunctionClass]map[oid.Oid][]function {
 			"crdb_internal.start_replication_stream",
 			"crdb_internal.replication_stream_progress",
 			"crdb_internal.complete_replication_stream",
+			"crdb_internal.revalidate_unique_constraint",
 		} {
 			skip = skip || strings.Contains(def.Name, substr)
 		}

